src/NewsService.Business/FindNewsCommand.cs

```diff
@@ -16,38 +34,186 @@ public class FindNewsCommand : IFindNewsCommand
+
+    private async Task<List<DepartmentData>> GetDepartments(List<Guid> departmentIds, List<string> errors)
+    {
+      if (departmentIds == null || departmentIds.Count == 0)
+      {
+        return null;
+      }
+
+      string errorMessage = "Can not get departments. Please try again later.";
+
+      try
+      {
+        Response<IOperationResult<IGetDepartmentsResponse>> response = await _rcGetDepartments
+          .GetResponse<IOperationResult<IGetDepartmentsResponse>>(IGetDepartmentsRequest.CreateObj(departmentIds));
+
+        if (response.Message.IsSuccess)
+        {
+          return response.Message.Body.Departments;
+        }
+        else
+        {
+          _logger.LogWarning("Errors while getting departments. Reason: {Errors}",
+            string.Join('\n', response.Message.Errors));
+        }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, errorMessage);
+      }
+
+      errors.Add(errorMessage);
+
+      return null;
+    }
 
     public FindNewsCommand(
       INewsRepository repository,
       INewsInfoMapper mapper,
-      IHttpContextAccessor httpContextAccessor)
+      IHttpContextAccessor httpContextAccessor,
+      IRequestClient<IGetDepartmentsRequest> rcGetDepartments,
+      IRequestClient<IGetUsersDataRequest> rcGetUsers,
+      IRequestClient<IGetImagesRequest> rcGetImages,
+      IDepartmentInfoMapper departmentInfoMapper,
+      ILogger<GetNewsCommand> logger,
+      IBaseFindRequestValidator baseFindValidator)
     {
       _repository = repository;
       _mapper = mapper;
       _httpContextAccessor = httpContextAccessor;
+      _rcGetDepartments = rcGetDepartments;
+      _rcGetUsers = rcGetUsers;
+      _rcGetImages = rcGetImages;
+      _logger = logger;
+      _baseFindValidator = baseFindValidator;
+      _departmentInfoMapper = departmentInfoMapper;
     }
 
-    public FindResultResponse<NewsInfo> Execute(FindNewsFilter findNewsFilter)
+    public async Task<FindResultResponse<NewsInfo>> Execute(FindNewsFilter findNewsFilter)
     {
       FindResultResponse<NewsInfo> response = new();
 
-      response.Body = _repository
-        .Find(findNewsFilter, out int totalCount)
-        .Select(_mapper.Map)
-        .ToList();
-
-      response.TotalCount = totalCount;
-
-      if (response.Errors.Any())
+      if (!_baseFindValidator.ValidateCustom(findNewsFilter, out List<string> errors))
       {
         _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
         response.Status = OperationResultStatusType.Failed;
+        response.Errors = errors;
         return response;
       }
 
+      List<DbNews> dbNewsList = _repository.Find(findNewsFilter, out int totalCount);
+      if (dbNewsList == null)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+
+        response.Errors = new() { "News was not found." };
+        response.Status = OperationResultStatusType.Failed;
```

Comment:
if dbNewsList == null i think it is return, add return

Reply:
done